app/models/annotation_text.rb

```diff
@@ -4,6 +4,7 @@ class AnnotationText < ApplicationRecord
   belongs_to :last_editor, class_name: 'User', foreign_key: :last_editor_id, optional: true
 
   after_update :update_mark_deductions,
+               if: :should_have_deduction?,
```

Comment:
I added this line because for the one-time annotations, I was getting an error on the next line (it was along the lines of nil does not have changes_to_save.key). So my Edit button was always failing on this line (causing the 500 Internal Server Error I talked about in my presentation last week). I noticed that the one-time annotations data being fetched didn't have deductions, so I added this check to only update_mark_deductions, if the annotation text had deductions.

Reply:
I see, good catch. Remove this line, but then down below explicitly check for whether `t.annotation_category` is `nil` (the check should be "unless the annotation category is nil or ...changes_to_save...").